### PR TITLE
# 5900 - Allow Markdown in Topic Descriptions

### PR DESCRIPTION
**Pull request by Miaplacidus**
This PR includes work that allows the user to add links (using HTML or Markdown syntax) to topic descriptions.

## Link to Issue
Closes: #5900 

## Description of Changes
- uses QuillRenderer to render topic descriptions

- Uses the Quill Editor to make the topic description
- Added validation for 250 character limit
- Links _should_ open a new tab
## Test Plan
- create a topic and add a link in its description using Markdown
- a clickable link should be present on the topic page

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/react_quill_editor/markdown_formatted_text.tsx [hunk 1]
```diff
@@ -176,7 +178,7 @@ export const MarkdownFormattedText = ({
         ref={containerRef}
         className={getClasses<{ collapsed?: boolean }>(
           { collapsed: isTruncated },
-          'MarkdownFormattedText',
+          customClass || 'MarkdownFormattedText',
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/react_quill_editor/markdown_formatted_text.tsx:181]: won't this break markdown format css if `customClass` is present?

**Comment by Israellund** [comment]
There are a few odd things happening that I found.
1. If you add an hmtl link like <a href="https://www.w3schools.com/">Visit W3Schools.com!</a> and click "Create thread" it doesn't show the link in the body of the thread, or anywhere for that matter. If I then put the same link into a comment, the link will show up inside the comment and is clickable, but it sends you to another page completely, it doesn't open a new tab. 
2. If you use a link like <a href="www.youtube.com">link text</a> (without the https://) it again won't show up in the body of the thread or anywhere else, but if you comment it it will show up as a clickable link. This time it won't open the link in any normal way, but instead the link is added to the breadcrumbs and shows you an error page. 
3. If you use an https:// address in a markdown link it doesn't open in another page, it redirects you (not sure if that's the desirable functionality). 
4. Sometimes when an https:// address is added to the alert popup in the markdown link it asks for the link again and will sometimes create the link twice. (see second video below)

Markdown 
https://github.com/hicommonwealth/commonwealth/assets/69872984/c8f9e3d3-8af5-4d53-a013-b174c2b12070


https://github.com/hicommonwealth/commonwealth/assets/69872984/8db4215a-2ca4-4f7a-a494-69dfaaed62ea

**Review by Israellund — APPROVED** [review]
LGTM

**Comment by lzach83** [comment]
I believe i have addressed all the above. Should be good for a look.

File: packages/commonwealth/client/scripts/state/api/topics/createTopic.ts [hunk 2]
```diff
@@ -21,6 +21,15 @@ const createTopic = async ({
...
 }: CreateTopicProps) => {
+  console.log(
+    'createTopic',
+    name,
+    description,
+    telegram,
+    featuredInSidebar,
+    featuredInNewPost,
+    defaultOffchainTemplate,
+  );
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/state/api/topics/createTopic.ts:32]: unnecesssary

**Review by kurtisassad — APPROVED** [review]
We should fix the dependency array eslint warning before merging, but otherwise looks good!

**Pull request merged**